apps/web/src/hooks/usePoolAvgInfo.ts

```diff
@@ -0,0 +1,74 @@
+/* eslint-disable no-console */
+import { ChainId } from '@pancakeswap/sdk'
+import { gql } from 'graphql-request'
+import useSWR from 'swr'
+
+import { v3Clients } from 'utils/graphql'
+
+export interface UsePoolAvgInfoParams {
+  numberOfDays?: number
+  address?: string
+  chainId?: ChainId
+}
+
+export const averageArray = (dataToCalculate: number[]): number => {
+  let data = [...dataToCalculate]
+  // Remove the highest and lowest volume to be more accurate
+  if (data.length > 3) {
+    data = data.sort((a: number, b: number) => a - b).slice(1, data.length - 1)
+  }
+
+  return data.reduce((result, val) => result + val, 0) / data.length
+}
+
+interface Info {
+  volumeUSD: number
+  tvlUSD: number
+  feesUSD: number
+}
+
+const defaultInfo: Info = {
+  volumeUSD: 0,
+  tvlUSD: 0,
+  feesUSD: 0,
+}
+
+export function usePoolAvgInfo({ address = '', numberOfDays = 7, chainId }: UsePoolAvgInfoParams) {
```

Comment:
do we need another subgraph request? I think we have similar one on useROI volume24H

Reply:
It's the same one